trl/trainer/grpo_trainer.py

```diff
@@ -1573,8 +1573,12 @@ def _generate_and_score_completions(
         completions_text = self.processing_class.batch_decode(completion_ids, skip_special_tokens=True)
         if is_conversational(inputs[0]):
             completions = []
-            for prompt, completion in zip(prompts, completions_text, strict=True):
-                bootstrap = prompt.pop()["content"] if prompt[-1]["role"] == "assistant" else ""
+            for prompt, completion in zip(prompts, completions_text):
```

Comment:
You should keep the change introduced by:
- #4183
```suggestion
            for prompt, completion in zip(prompts, completions_text, strict=True):
```